python/hsfs/feature_view.py

```diff
@@ -515,9 +529,73 @@ def get_feature_vectors(
         """
         if self._batch_vectors_server is None:
             self.init_serving(external=external)
+        updated_passed_feature = []
+        for i in range(len(entry)):
+            updated_passed_feature.append(self._update_with_vector_db_result(
+                self._batch_vectors_server,
+                entry[i], passed_features[i] if passed_features else {}
+            ))
         return self._batch_vectors_server.get_feature_vectors(
-            entry, return_type, passed_features, allow_missing
+            entry, return_type, updated_passed_feature, allow_missing
+        )
+
+    def _update_with_vector_db_result(self, vec_server, entry, passed_features):
+        for k, fg in self._vector_db_client.embedding_fg_by_join_index.items():
+            complete, fg_entry = vec_server.filter_entry_by_join_index(
+                entry, k
+            )
+            if not complete:
+                # Not retrieving from vector db if entry is not completed
+                continue
+            vector_db_features = self._vector_db_client.read(
+                fg.id, keys=fg_entry, index_name=fg.embedding.index_name
+            )
+
+            # if result is not empty
+            if vector_db_features:
+                vector_db_features = vector_db_features[0]  # get the first result
+                if passed_features and vector_db_features:
+                    vector_db_features.update(passed_features)
+                    passed_features = vector_db_features
+                else:
+                    passed_features = vector_db_features
+        return passed_features
+
+
+    def find_neighbors(self, embedding, feature=None, k=10, filter=None, min_score=0):
+        if self._vector_db_client is None:
+            self.init_serving()
+        results = self._vector_db_client.find_neighbors(
+            embedding, feature=(feature if feature else None),
+            k=k, filter=filter, min_score=min_score
         )
+        if len(results) == 0:
+            return []
+
+        passed_features = []
+        for res in results:
+            passed_features.append(res[1])
+        return self.get_feature_vectors(
+            [self._extract_primary_key(res[1]) for res in results],
+            passed_features=passed_features,
+            external=True,
+            allow_missing=True
+        )
+
+    def _extract_primary_key(self, result_key):
+        primary_key_map = {}
+        for prefix_sk, sk in self._prefix_serving_key_map.items():
+            if prefix_sk in result_key:
+                primary_key_map[sk.required_serving_key] = result_key[prefix_sk]
```

Comment:
In this case, it needs look up twice anyway because result of `prefix_sk in result_key` and `sk.feature_name in result_key` needs to be saved beforehand.